bukkit/rpk-unconsciousness-bukkit/src/main/kotlin/com/rpkit/unconsciousness/bukkit/listener/PlayerJoinListener.kt

```diff
@@ -34,9 +34,9 @@ class PlayerJoinListener : Listener {
         val minecraftProfileService = Services[RPKMinecraftProfileService::class.java] ?: return
         val characterService = Services[RPKCharacterService::class.java] ?: return
         val unconsciousnessService = Services[RPKUnconsciousnessService::class.java] ?: return
-        val minecraftProfile = minecraftProfileService.getPreloadedMinecraftProfile(bukkitPlayer) ?: return
-        val character = characterService.getPreloadedActiveCharacter(minecraftProfile) ?: return
-        if (unconsciousnessService.isUnconscious(character)) return
+        val minecraftProfile = minecraftProfileService.getMinecraftProfile(bukkitPlayer) ?: return
+        val character = characterService.getActiveCharacter(minecraftProfile) ?: return
+        if (unconsciousnessService.isUnconscious(character).join()) return
```

Comment:
PlayerJoinEvent is fired synchronously, don't use join.
If you use thenAccept remember to remove potion effect in a sync task. Alternatively use preloaded unconsciousness state in unconsciousness service.